### PR TITLE
Key Storage through Enterprise Runner

### PR DESCRIPTION
**Pull request by jsboak**

### CODE REVIEW

File: docs/manual/key-storage/enterprise-runner-key-storage.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+As of version **`4.16.0`**, integration with Hashicorp Vault is supported on the Runner.<br>
+In future releases, integration with **CyberArk** and **Thycotic** will also be available through the Runner.
+:::
+
+## Use Cases
+1. **Remote Node Commands & Scrips** (Node Executors & File Copiers): When SSH or WinRM credentials are stored in a secrets provider, then the Runner can retrieve keys from the provider to authenticate with remote nodes in order to execute commands or scripts.<br><br>
+2. **Internal Tools & Infrastructure** (Job step plugins): When executing Jobs that include steps that integrate with internal tools APIs - such as Jira, Jenkins, homegrown tooling, etc. - or infrastructure such as databases, then the Runner can use secrets to authenticate with these endpoints using best-practice security standards.<br><br>
+3. **Inventory Discovery** (Node Sources): The Runner can be used to discover inventory in secure or remote environments. By retrieving keys from a secrets-provider, the Runner can authenticate with an API endpoint, such as the VMware vSphere API, in order to retrieve the inventory.
+
+:::warning Support for Node Discovery and Remote Commands
```
Comment (fdevans) [inline docs/manual/key-storage/enterprise-runner-key-storage.md]: I think this warning needs to be clearer (more words) to define the limitations and maybe made into it's own paragraph or section of the page called "Current Limitations".  I could go after the Guardrails section so there is context for why certain things aren't supported.

File: docs/manual/key-storage/enterprise-runner-key-storage.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+1. Secrets retrieved by a Runner can only be used for operations executed on that Runner.  These secrets are not sent back to the Process Automation cluster or Runbook Automation instance.  Therefore, these secrets can not be used for plugins that do not reside on the Runner.<br><br>
+2. Secrets are masked in logs, including Job execution logs.  Any secrets retrieved by the Runner and printed to the Job log output will appear as `[SECURE]` in the logs.<br><br>
+3. Sensitive environment variables set on the Runner are masked in logs, including Job execution logs.  Any environment variables ending with `TOKEN` or `KEY` or `PASSWORD` will be printed as `[SECURE]` in the logs.
+
+## Configuration
+To configure a Key Storage integration on the Runner, **configuration properties are set on the Runner**. These properties can be set through the following methods:
+
+**Note:** These examples do not include the full list of configuration properties to configure an integration with Vault.  The full list of
+configuration properties can be found [here](/manual/key-storage/storage-plugins/vault.html#configuration).
+
```
Comment (fdevans) [inline docs/manual/key-storage/enterprise-runner-key-storage.md:37]: For the three examples I think we should have a recommended approach between YAML and Env variable.

If they do YAML and don't include other settings the Runner needs to operate (url for example) will it still work?

File: docs/manual/key-storage/enterprise-runner-key-storage.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+   ![Runner Select Key](@assets/img/runner-select-key.png)
+8. Click on **Choose Selected Key**.
+9. In the Job step, the field will now be populated with `runner/path/to/key`:
+    ![Job Step Runner Key](@assets/img/job-step-using-runner-key.png)
+
+:::warning Using Secrets from Runners in Jobs
+* In order to use secrets retrieved by a Runner within Jobs, the Runner **Tag** used for secrets must align with **Tag**
+used for the **Runner Selector** in the **Nodes & Runner** tab of the Job configuration.
+
+* If the selected tag used for **Nodes & Runners** and the key storage browser contains more than one Runner, then all Runners with that tag must be integrated with the secrets-management provider.
```
Comment (fdevans) [inline docs/manual/key-storage/enterprise-runner-key-storage.md]: Is this to imply "then all Runners with that tag must be integrated with the **_SAME_** secrets-management provider"

File: docs/manual/key-storage/enterprise-runner-key-storage.md (hunk 1 above)
Comment (jsboak) [reply docs/manual/key-storage/enterprise-runner-key-storage.md]: you're saying after the **Handling Secrets** section?

File: docs/manual/key-storage/enterprise-runner-key-storage.md (hunk 3 above)
Comment (jsboak) [reply docs/manual/key-storage/enterprise-runner-key-storage.md]: Yes - i will rephrase

File: docs/manual/key-storage/enterprise-runner-key-storage.md (hunk 2 above)
Comment (jsboak) [reply docs/manual/key-storage/enterprise-runner-key-storage.md:37]: Why do we need to be opinionated about whether customers use YAML vs. Env variable?

And yes: the YAML can include just the key storage config properties.  the connection properties for communicating with the server are defined elsewhere.

Comment (fdevans) [reply docs/manual/key-storage/enterprise-runner-key-storage.md:37]: Given multiple options how do they decide which to use?

Comment (jsboak) [reply docs/manual/key-storage/enterprise-runner-key-storage.md:37]: resolved in DM

**Pull request merged**